### ISSUE TITLE
HMAC-SHA256

### ISSUE THREAD

**Issue by gillesdemey** [request/oauth-sign#13]
Would you accept a PR that adds the SHA256 hashing algorithm?

It's not officially part of the specification, but the spec has the following to say regarding signature methods:

```
OAuth does not mandate a particular signature method, as each
implementation can have its own unique requirements.  Servers are
free to implement and document their own custom methods.
```

I understand that it's probably a very unique requirement on our part, I'm just curious if you would accept such changes upstream and if it would make it into the request module.

**Comment by jsmartfo** [comment]
I am also looking for this feature... @gillesdemey what was your workaround when using request module? 
Thanks

**Comment by jsmartfo** [comment]
I have this change working as @gillesdemey notes above will you accept a PR for this? 

I am also looking for support for the request module.

**Comment by gillesdemey** [comment]
I had a working fork of this package where I implemented the SHA256 algorithm, but we have since made some ajustments on the server to be compliant with the industry's standards.

I can't seem to find it, but re-implementing this is trivial so if there's any interest in merging I can just whip it up again.

I'm also willing to test and review a PR, if this is something you're willing to merge upstream.

**Comment by kavimadhu** [comment]
I am also looking for this implementation. So what did you to fix this in your application?

**Comment by jsmartfo** [comment]
The change that worked for me are in a fork here; 

https://github.com/jsmartfo/oauth-sign/commit/70de9c712bf20fe2f4216b2aedde77977eae0c38

Would be willing to cleanup accordingly

**Comment by kylelaverty** [comment]
@jsmartfo Is there any chance this can be added to this repo?

I am looking to use this project but need the ability to support HMAC-SHA256.

**Comment by kylelaverty** [comment]
I ended up just creating a PR from my fork for this support - (PR#23)[https://github.com/request/oauth-sign/pull/23]

**Comment by simov** [comment]
Published in version 0.9.0
